crates/codegen/language/definition/src/model/terminals/keyword.rs

```diff
@@ -32,6 +32,23 @@ pub enum KeywordValue {
     Atom { atom: String },
 }
 
+impl From<KeywordValue> for Scanner {
```

Comment:
The `KeywordValue` here exists specifically to differentiate itself from a `Scanner`, as it has different/fewer operations, because of how we want to enforce an FSM model in the future, and how we expect keyword reservation to work.

I understand if such conversion makes sense for the specific parser codegen we have now/if it is handled the same, but I'm a bit hesitant about introducing a common utility in the DSL crate to convert it to a scanner, because that implies that a scanner can be used instead. I wonder if we should keep such conversion in the parser codegen instead?

Just a nit suggestion. Not blocking of course.